x-pack/plugins/security/server/routes/api/public/roles/put.js

```diff
@@ -9,20 +9,35 @@ import Joi from 'joi';
 import { ALL_RESOURCE } from '../../../../../common/constants';
 import { wrapError } from '../../../../lib/errors';
 
-const transformKibanaPrivilegeToEs = (application, kibanaPrivilege) => {
-  return {
-    privileges: kibanaPrivilege.privileges,
-    application,
-    resources: [ALL_RESOURCE],
-  };
+const transformKibanaPrivilegesToEs = (application, kibanaPrivileges) => {
+  const kibanaApplicationPrivileges = [];
+  if (kibanaPrivileges.global && kibanaPrivileges.global.length) {
+    kibanaApplicationPrivileges.push({
+      privileges: kibanaPrivileges.global,
+      application,
+      resources: [ALL_RESOURCE],
+    });
+  }
+
+  if (kibanaPrivileges.space) {
+    for(const [spaceId, privileges] of Object.entries(kibanaPrivileges.space)) {
```

Comment:
I like this! I always forget about `Object.entries`, and the fact that you can destructure arrays like this

Reply:
I usually forget also, but I remembered this time around :)